src/cpp/fastdds/domain/DomainParticipantFactory.cpp

```diff
@@ -339,6 +392,18 @@ ReturnCode_t DomainParticipantFactory::set_qos(
     return ReturnCode_t::RETCODE_OK;
 }
 
+void DomainParticipantFactory::reset_default_participant_qos()
+{
+    // TODO (Miguel C): Change when we have full XML support for DDS QoS profiles
+    DomainParticipantImpl::set_qos(default_participant_qos_, PARTICIPANT_QOS_DEFAULT, true);
```

Comment:
If `default_xml_profiles_loaded` is true, the whole `default_participant_qos_` will be overridden with the values in the attributes. This instruction can be relegated to the `else` clause.

Reply:
Not exactly true, as entity_factory is not loaded from XML

Reply:
So you want to keep the entity_factory from the default QoS. :+1: